src/libcore/str/lossy.rs

```diff
@@ -146,7 +147,11 @@ impl<'a> Iterator for Utf8LossyChunksIter<'a> {
             broken: &[],
         };
         self.source = &[];
-        return Some(r);
+        Some(r)
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (0, Some(self.source.len()))
```

Comment:
The idea here is that we could have 0 due to an error, and `self.source.len()` if every character is ASCII?

Side note: I wonder if we should have small comments on all these describing the logic and rationale behind them, just like we encourage for `unsafe` blocks.

Reply:
Yes, that's the idea. I'll add a comment about it.